src/java/org/apache/cassandra/index/sai/disk/v1/SSTableIndexWriter.java

```diff
@@ -394,53 +400,55 @@ else if (indexContext.isLiteral())
 
     private static boolean allRowsHaveVectorsInWrittenSegments(IndexContext indexContext)
     {
-        for (SSTableIndex index : indexContext.getView().getIndexes())
+        // TODO should we load all of these for this op? It seems very unlikely that we want to consider the whole
+        // table when checking if all rows have vectors. A single empty vector will be enough to make this false,
+        // but that should really only impact that table.
+        var view = indexContext.getReferencedView(TimeUnit.SECONDS.toNanos(5));
```

Comment:
I am moving this TODO out of scope for this PR. Here is a follow up ticket: https://github.com/riptano/cndb/issues/14028